### PR TITLE
(WIP) Sgv2 #1435: Add createTable()

### LINKED ISSUE #1435: Implement Stargate V2 REST endpoints for "createTable" using new gRPC endpoint
(similar to #1426)

Stargate V2 REST implementation has accessor for Table metadata but "createTable()" missing -- Integration tests for now use SGv1 version. Implementing creation natively is possibly the last missing piece needed for full end-to-end testing of the minimal functionality (create ks, table(s); insert data; get data).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Implements "createTable" for Tables Resource using SGv2 mechanism (going via gRPC bridge).

For some reason there seem to be issues in conjunction with keyspace creation on tests.

**Which issue(s) this PR fixes**:
Fixes #1435

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated

### CODE REVIEW

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java [hunk 1]
```diff
@@ -185,14 +187,34 @@ public Response createTable(
...
-    logger.info("createTable() called for KS '" + keyspaceName + "'");
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("keyspaceName must be non-empty").build();
+    }
+    final String tableName = tableAdd.getName();
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("Table name must be non-empty").build();
+    }
+    final StargateGrpc.StargateBlockingStub blockingStub =
+        grpcFactory.constructBlockingStub().withCallCredentials(new StargateBearerToken(token));
```
Comment (dougwettlaufer) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:203]: When we start looking at performance we might want to consider using async.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java [hunk 2]
```diff
@@ -185,14 +187,34 @@ public Response createTable(
...
     if (isAuthTokenInvalid(token)) {
       return invalidTokenFailure();
     }
-    logger.info("createTable() called for KS '" + keyspaceName + "'");
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("keyspaceName must be non-empty").build();
+    }
+    final String tableName = tableAdd.getName();
+    if (isStringEmpty(keyspaceName)) {
+      return jaxrsBadRequestError("Table name must be non-empty").build();
```
Comment (dougwettlaufer) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:200]: Can we retain the existing error for this and the keyspace?

```java
return Response.status(Response.Status.BAD_REQUEST)
                .entity(
                    new ApiError(
                        "table name must be provided", Response.Status.BAD_REQUEST.getStatusCode()))
                .build();
```

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java [hunk 3]
```diff
@@ -287,6 +309,46 @@ public Response deleteTable(
...
+      QueryOuterClass.ColumnSpec column =
+          QueryOuterClass.ColumnSpec.newBuilder()
+              .setName(columnName)
+              .setType(
+                  BridgeProtoTypeTranslator.cqlTypeFromBridgeTypeSpec(
+                      columnDef.getTypeDefinition()))
+              .build();
+      if (primaryKeys.hasClusteringKey(columnName)) {
+        b.addClusteringKeyColumns(column);
+      } else if (primaryKeys.hasClusteringKey(columnName)) {
```
Comment (dougwettlaufer) [inline sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:329]: Should be `hasPartitionKey`?

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java (hunk 1 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:203]: Yes, absolutely.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java (hunk 2 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:200]: Will do.

File: sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java (hunk 3 above)
Comment (tatu-at-datastax) [reply sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/schemas/Sgv2TablesResource.java:329]: Yes. Good catch.

**Pull request merged**